homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -411,16 +657,16 @@ async def async_stop_cover_tilt(self, **kwargs):
     @callback
     def _async_process_updated_position_data(self, position_data):
         """Process position data."""
-        if ATTR_POSKIND1 not in position_data:
-            return
-        if int(position_data[ATTR_POSKIND1]) == POSKIND_PRIMARY:
-            self._current_hd_cover_position = int(position_data[ATTR_POSITION1])
-            self._attr_current_cover_tilt_position = 0
-        if int(position_data[ATTR_POSKIND1]) == POSKIND_VANE:
-            self._current_hd_cover_position = MIN_POSITION
-            self._attr_current_cover_tilt_position = hd_position_to_hass(
-                int(position_data[ATTR_POSITION1]), self._max_tilt
-            )
+        # shades must be closed to tilt and vane must be closed to lift
+        # shade position needs to be assumed as only 1 poskind is returned
+        # this is not a guess, it is impossible for any other state
+        if ATTR_POSITION1 in position_data:
+            if position_data[ATTR_POSKIND1] == POS_KIND_PRIMARY:
+                self.set_position_primary(position_data[ATTR_POSITION1])
+                self.set_position_vane(MIN_POSITION)
+            if position_data[ATTR_POSKIND1] == POS_KIND_VANE:
+                self.set_position_primary(MIN_POSITION)
+                self.set_position_vane(position_data[ATTR_POSITION1])
```

Comment:
```suggestion
        if (position1 := position_data.get(ATTR_POSITION1)) is None:
            return
        position1_kind = position_data[ATTR_POSKIND1]
        if position1_kind == POS_KIND_PRIMARY:
            self.set_position_primary(position1)
            self.set_position_vane(MIN_POSITION)
        elif position1_kind == POS_KIND_VANE:
            self.set_position_primary(MIN_POSITION)
            self.set_position_vane(position1)
```

Assuming ATTR_POSITION1 is never `None` you could write it like this.